mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxStateStore.kt

```diff
@@ -72,6 +78,8 @@ internal open class MvRxStateStore<S : Any>(initialState: S) : Disposable {
      * all of the code required.
      */
     fun set(stateReducer: S.() -> S) {
+        if (MvRxMocker.getMockedState(this) != null) return
```

Comment:
do you see reasons to only block this if mocked state is set? I think it should be just
```
if (isMocked)
```

Because the view model or fragment might do things to set state before the mock is in place (the mock can only be set once the view model is created). Since setting state is async and can trigger subscribe callbacks it would be nice to eliminate any side effects from that